pkg/monitoring/multi_feed_monitor.go

```diff
@@ -49,67 +49,102 @@ func (m *multiFeedMonitor) Run(ctx context.Context, data RDDData) {
 	var subs utils.Subprocesses
 	defer subs.Wait()
 
-FEED_LOOP:
+	// iterate over each feed
 	for _, feedConfig := range data.Feeds {
 		feedLogger := logger.With(m.log,
 			"feed_name", feedConfig.GetName(),
 			"feed_id", feedConfig.GetID(),
 			"network", m.chainConfig.GetNetworkName(),
 		)
-		// Create data sources
-		pollers := []Poller{}
+
+		// create sources outside of createMonitor
+		sources := []Source{}
+		sourceTypes := []string{}
 		for _, sourceFactory := range m.sourceFactories {
 			source, err := sourceFactory.NewSource(m.chainConfig, feedConfig)
 			if err != nil {
 				feedLogger.Errorw("failed to create source", "error", err, "source-type", fmt.Sprintf("%T", sourceFactory))
 				continue
 			}
-			poller := NewSourcePoller(
-				source,
-				logger.With(m.log, "component", "chain-poller", "source", sourceFactory.GetType()),
-				m.chainConfig.GetPollInterval(),
-				m.chainConfig.GetReadTimeout(),
-				m.bufferCapacity,
-			)
-			pollers = append(pollers, poller)
-		}
-		if len(pollers) == 0 {
-			feedLogger.Errorw("not tracking feed because all sources failed to initialize")
-			continue FEED_LOOP
+			sources = append(sources, source)
+			sourceTypes = append(sourceTypes, sourceFactory.GetType())
 		}
-		// Create exporters
-		exporters := []Exporter{}
-		for _, exporterFactory := range m.exporterFactories {
-			exporter, err := exporterFactory.NewExporter(ExporterParams{
+
+		createMonitor(
+			ctx,
+			&subs,
+			feedLogger,
+			m.chainConfig,
+			m.bufferCapacity,
+			sources,
+			sourceTypes,
+			m.exporterFactories,
+			ExporterParams{
 				m.chainConfig,
 				feedConfig,
 				data.Nodes,
-			})
-			if err != nil {
-				feedLogger.Errorw("failed to create new exporter", "error", err, "exporter-type", fmt.Sprintf("%T", exporterFactory))
-				continue
-			}
-			exporters = append(exporters, exporter)
-		}
-		if len(exporters) == 0 {
-			feedLogger.Errorw("not tracking feed because all exporters failed to initialize")
-			continue FEED_LOOP
-		}
-		// Run poller goroutines.
-		for _, poller := range pollers {
-			poller := poller
-			subs.Go(func() {
-				poller.Run(ctx)
-			})
-		}
-		// Run feed monitor.
-		feedMonitor := NewFeedMonitor(
-			logger.With(m.log, "component", "feed-monitor"),
-			pollers,
-			exporters,
+			},
 		)
+	}
+}
+
+// createMonitor is a reusable method for creating pollers, exporters, and monitors
+// sources are created outside because they may require different input parameters
+func createMonitor(
```

Comment:
WDYT of renaming this to `runMonitor()`. Also since this is a shared function, how about moving it to a new file? maybe `run_monitor.go`.